docs: Refactored product storage status into eo product properties (#1612)
CS-SI/eodag: docs/getting_started_guide/eo_products_properties.rst

Reviewer: > - [ ] add `id` as _Unique identifier for the resource, harmonized across providers._
> 
`id` is still missing

Reviewer: please sort properties in alphabetical order

Reviewer: please add ` ` double ticks around properties values to make them more visible (`VV`,`VV VH`, ...)

.. _eo_products_properties:

EO Products Properties
======================

> EODAG exposes metadata for queryables based on [OGC OpenSearch Extension for Earth Observation](https://docs.ogc.org/is/13-026r9/13-026r9.html) for all providers that support it.
> All of the parameters are string based. If a parameter requires an specific format it will be defined on the list below.
> 
> ----------------------------

Reviewer: ```suggestion
EODAG maps each provider's specific metadata parameters to a common model using
`OGC OpenSearch Extension for Earth Observation <http://docs.opengeospatial.org/is/13-026r9/13-026r9.html>`_.
Extra parameters that have no equivalent in this model are mapped as they are.
For more information on parameters mapping, please see `Contributing / Parameters mapping <params_mapping.rst>`_.

Here is a list of these common mapped properties:
```

> - **uid**: Unique identifier for the resource.

Reviewer: ```suggestion
- **uid**: Unique identifier for the resource, using provider specific formatting.
```

- **keyword**: Commonly used word(s) or formalized word(s) or phrase(s) used to describe the subject.
- **abstract**: A brief narrative summary of the resource.
- **resolution**: The spatial resolution of the resource.
- **organisationName**: Name of the organization responsible for the resource.
- **organisationRole**: Role of the organization (e.g., resourceProvider, custodian, owner, user, distributor, originator, pointOfContact, principalInvestigator, processor, publisher, author).
- **publicationDate**: Date when the resource was published.
- **lineage**: Information about the lineage of the resource.
- **useLimitation**: Constraints on the usage of the resource.
- **accessConstraint**: Restrictions and legal prerequisites for accessing the resource.
- **otherConstraint**: Other constraints related to the resource.
- **classification**: Security classification of the resource (e.g., unclassified, restricted, confidential, secret, topSecret).
- **language**: Language of the resource content.
- **specification**: Reference to the specification to which the resource conforms.
- **parentIdentifier**: A string identifying the parent of the entry in a hierarchy of resources.

> - **productionStatus**: A string identifying the status of the entry (e.g., ARCHIVED, ACQUIRED, CANCELLED).

Reviewer: ```suggestion
```
Should be removed, see https://eodag.readthedocs.io/en/latest/changelog.html#v1-5-1-2020-04-08

Please also remove remaining occurrences of `productionStatus` in the code

- **processingCenter**: Center responsible for processing the product.
- **creationDate**: Date when the product was created.
- **modificationDate**: Date when the product was last modified.
- **processingDate**: Date when the product was processed.
- **sensorMode**: Mode of the sensor during acquisition.
- **archivingCenter**: Center responsible for archiving the product.
- **processingMode**: Mode used during processing.
- **availabilityTime**: The time when the result became available (e.g., if post-processing or laboratory analysis is required, it might differ from the phenomenon time). Format: ISO 8601 dateTime.
- **acquisitionStation**: A string identifying the station used for the acquisition.
- **acquisitionSubType**: Acquisition sub-type.
- **startTimeFromAscendingNode**: Start time from the ascending node.
- **completionTimeFromAscendingNode**: Completion time from the ascending node.
- **illuminationAzimuthAngle**: Illumination azimuth angle during acquisition.
- **illuminationZenithAngle**: Illumination zenith angle during acquisition.
- **illuminationElevationAngle**: Illumination elevation angle during acquisition.

> - **polarizationMode**: Mode of polarization, space-separated (i.e. VV, VV VH, etc).

Reviewer: ```suggestion
- **polarizationMode**: Mode of polarization.
```

> - **polarizationChannels**: Channels used for polarization.

Reviewer: ```suggestion
- **polarizationChannels**: Channels used for polarization, space-separated (i.e. ``"VV"``, ``"VV VH"``, etc).
```

.. _eo_products_properties:

EO Products Properties
======================

EODAG exposes metadata for queryables based on [OGC OpenSearch Extension for Earth Observation](https://docs.ogc.org/is/13-026r9/13-026r9.html) for all providers that support it.
All of the parameters are string based. If a parameter requires an specific format it will be defined on the list below.

----------------------------

* **uid**: Unique identifier for the resource, using provider specific formatting.
* **tileIdentifier**: Tile identifier.

> * **productType**: A string identifying the entry type (e.g., ER02_SAR_IM__0P, MER_RR__1P, SM_SLC__1S, GES_DISC_AIRH3STD_V005).

Reviewer: please use product types existing in `resources/product_types.yml` as example (`S2_MSI_L1C`, `LANDSAT_C2L2_SR`, `CAMS_EAC4`, ...)

* **doi**: Digital Object Identifier identifying the product.

> * **platform**: A string with the platform short name (e.g., Sentinel-1).

Reviewer: ```suggestion
* **platform**: A string with the platform short name (e.g., SENTINEL1).
```

Reviewer: see format in `eodag/resources/product_types.yml`

* **abstract**: A brief narrative summary of the resource.
* **resolution**: The spatial resolution of the resource.
* **organisationName**: Name of the organization responsible for the resource.
* **organisationRole**: Role of the organization (e.g., resourceProvider, custodian, owner, user, distributor, originator, pointOfContact, principalInvestigator, processor, publisher, author).
* **publicationDate**: Date when the resource was published.
* **lineage**: Information about the lineage of the resource.
* **useLimitation**: Constraints on the usage of the resource.
* **accessConstraint**: Restrictions and legal prerequisites for accessing the resource.
* **otherConstraint**: Other constraints related to the resource.
* **classification**: Security classification of the resource (e.g., unclassified, restricted, confidential, secret, topSecret).
* **language**: Language of the resource content.
* **specification**: Reference to the specification to which the resource conforms.
* **parentIdentifier**: A string identifying the parent of the entry in a hierarchy of resources.
* **productionStatus**: A string identifying the status of the entry (e.g., ARCHIVED, ACQUIRED, CANCELLED).

> * **acquisitionType**: Used to distinguish at a high level the appropriateness of the acquisition for general use. Values: NOMINAL, CALIBRATION, OTHER.

Reviewer: please do not list possible values as we do not harmonize them for the moment

* **orbitNumber**: A number, set, or interval requesting the acquisition orbit.

> * **orbitDirection**: A string identifying the acquisition orbit direction. Possible values: ASCENDING, DESCENDING.

Reviewer: please do not list possible values as we do not harmonize them for the moment

* **completionTimeFromAscendingNode**: Completion time from the ascending node.
* **illuminationAzimuthAngle**: Illumination azimuth angle during acquisition.
* **illuminationZenithAngle**: Illumination zenith angle during acquisition.
* **illuminationElevationAngle**: Illumination elevation angle during acquisition.
* **polarizationMode**: Mode of polarizationChannel.
* **polarizationChannels**: Channels used for polarization, space-separated (i.e. VV, VV VH, etc).
* **antennaLookDirection**: Direction in which the antenna was looking during acquisition.
* **minimumIncidenceAngle**: Minimum incidence angle during acquisition.
* **maximumIncidenceAngle**: Maximum incidence angle during acquisition.
* **dopplerFrequency**: Doppler frequency during acquisition.
* **incidenceAngleVariation**: Variation in incidence angle during acquisition.
* **storageStatus**: Storage status on the provider. One of the following strings:

  * ONLINE: the product is available for download (immediately);
  * STAGING: the product has been ordered and will be `ONLINE` soon;
  * OFFLINE: the product is not available for download, but can eventually be ordered.

>     * ``eodag`` is able to order `OFFLINE` products and retry downloading them for a while. This is described in more details in the `Python API user guide <../notebooks/api_user_guide/8_download.ipynb>`_.

Reviewer: bad RST formatting
